src/renderer/cluster-frame.tsx

```diff
@@ -120,9 +124,14 @@ export class ClusterFrame extends React.Component {
 
     whatInput.ask(); // Start to monitor user input device
 
+    const clusterContext = new FrameContext(cluster);
+
     // Setup hosted cluster context
     KubeObjectStore.defaultContext.set(clusterContext);
-    kubeWatchApi.context = clusterContext;
+    WorkloadsOverview.clusterContext
+      = KubeObjectListLayout.clusterContext
+      = KubeWatchApi.context
+      = clusterContext;
```

Comment:
Why have clusterContext now stored in 3 places instead of one?

Reply:
Because each they all individually need to know about the selected namespaces.

Reply:
But they could already do that via import, right?

Reply:
I guess so, I guess this is a form of dependency injection.